ui/src/layouts/BuildPublication.tsx

```diff
@@ -60,9 +60,29 @@ const BuildPublication: React.FC<BuildPublicationProps> = (props) => {
                     message: 'You have selected there is a conflict of interest, please provide a reason.'
                 };
             }
-            if (typeof store.conflictOfInterestStatus == 'undefined') {
+
+            if (typeof store.conflictOfInterestStatus === 'undefined') {
                 ready = { ready: false, message: 'You must select a conflict of interest option' };
             }
+
+            if (store.affiliationsStatus && !store.affiliations.length) {
```

Comment:
you also need to add these validations on the backend as well, because currently the user can publish even if they selected 'Yes' but didn't add any affiliations at all. The opposite is true as well, they can publish even if they selected 'No' but they didn't confirm they have no affiliations.